dashboard/http_server_head.py

```diff
@@ -145,6 +182,39 @@ async def run(self, modules):
             logger.info(r)
         logger.info("Registered %s routes.", len(dump_routes))
 
+    def _setup_metrics(self, app):
+        metrics = DashboardPrometheusMetrics()
+        app["metrics"] = metrics
+        app.middlewares.insert(0, self.metrics_middleware)
+
+        # Setup prometheus metrics export server
+        _initialize_internal_kv(self.gcs_client)
+        address = f"{self.head_node_ip}:{DASHBOARD_METRIC_PORT}"
+        _internal_kv_put("DashboardMetricsAddress", address, True)
+        if prometheus_client:
+            try:
+                logger.info(
+                    "Starting dashboard metrics server on port {}".format(
+                        DASHBOARD_METRIC_PORT
+                    )
+                )
+                kwargs = (
+                    {"addr": "127.0.0.1"} if self.head_node_ip == "127.0.0.1" else {}
```

Comment:
For the host, why don't we reuse this part? (line 133 in this file)

```
        self.http_host, self.http_port, *_ = site._server.sockets[0].getsockname()
        self.http_host = (
            self.ip
            if ipaddress.ip_address(self.http_host).is_unspecified
            else self.http_host
        )
        logger.info(
            "Dashboard head http address: %s:%s", self.http_host, self.http_port
        )
```

Reply:
can you explain more? What can we re-use here?